once_test.py

```diff
@@ -729,6 +823,29 @@ async def counting_fn2():
         self.assertEqual(await counting_fn2(), 2)
         self.assertEqual(await counting_fn2(), 2)
 
+    async def test_once_per_thread(self):
+        counter = Counter()
+
+        @once.once(per_thread=True)
+        async def counting_fn(*args) -> int:
+            """Returns the call count, which should always be 1."""
+            nonlocal counter
+            del args
+            return counter.get_incremented()
+
+        @execute_with_barrier(n_workers=_N_WORKERS)
+        def execute():
+            result = counting_fn()
```

Comment:
nit: result is a confusing name here, since it really just holds a coroutine.  I'd either go with `coro = ` or preferably just inline it as `return asyncio.run(counting_fn())`

Reply:
done